plugin/src/main/java/io/jenkins/plugins/analysis/core/steps/WarningChecksPublisher.java

```diff
@@ -40,10 +40,16 @@
 class WarningChecksPublisher {
     private final ResultAction action;
     private final TaskListener listener;
+    private final String checksName;
 
-    WarningChecksPublisher(final ResultAction action, final TaskListener listener) {
+    WarningChecksPublisher(final ResultAction action, final TaskListener listener, final String checksName) {
```

Comment:
See above: parameter should be of type `ChecksInfo`. 
And do not add a second constructor since we need the same call for the `IssueRecorder` (which is used by the `recordIssues` step).